app/src/main/java/com/android/shelfLife/viewmodel/recipes/RecipeGenerationViewModel.kt

```diff
@@ -0,0 +1,113 @@
+package com.android.shelfLife.viewmodel.recipes
+
+import android.util.Log
+import androidx.compose.runtime.toMutableStateList
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.android.shelfLife.model.newFoodItem.FoodItem
+import com.android.shelfLife.model.newFoodItem.FoodItemRepository
+import com.android.shelfLife.model.newRecipe.RecipeRepository
+import com.android.shelfLife.model.recipe.Recipe
+import com.android.shelfLife.model.recipe.RecipeGeneratorRepository
+import com.android.shelfLife.model.recipe.RecipePrompt
+import com.android.shelfLife.model.user.UserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+open class RecipeGenerationViewModel
```

Comment:
There seems to be no error handling for the text inputs here. Check out the utils functions cause they do error handling for things like name strings

Reply:
I was doing te validation inside the Screen actually, so that I could show toast errors.
- I now used the validateString like you described